### PR TITLE
Rename `DatasetEventManager` to `DatasetManager`

### PR DESCRIPTION
**Pull request by jedcunningham**
This manager class will end up dealing with more than just dataset
events over time, so rename it to be more general.

### CODE REVIEW

File: airflow/config_templates/config.yml [hunk 1]
```diff
@@ -386,13 +386,13 @@
       type: string
       default: "0o077"
       example: ~
-    - name: dataset_event_manager_class
+    - name: dataset_manager_class
       description: Class to use as dataset event manager.
```
Comment (blag) [inline airflow/config_templates/config.yml]:
```suggestion
      description: Class to use as dataset manager, including dataset events.
```

File: airflow/config_templates/config.yml [hunk 2]
```diff
@@ -386,13 +386,13 @@
...
       version_added: 2.4.0
       type: string
       default: ~
-      example: 'airflow.datasets.manager.DatasetEventManager'
-    - name: dataset_event_manager_kwargs
+      example: 'airflow.datasets.manager.DatasetManager'
+    - name: dataset_manager_kwargs
       description: Kwargs to supply to dataset event manager.
```
Comment (blag) [inline airflow/config_templates/config.yml]:
```suggestion
      description: Kwargs to supply to dataset manager.
```

**Review by blag — CHANGES_REQUESTED** [review]
Looking pretty good, just a few more places to fix.

File: airflow/config_templates/config.yml (hunk 1 above)
Comment (jedcunningham) [reply airflow/config_templates/config.yml]: I think I'm 👎 on this specific one.

File: airflow/config_templates/config.yml [hunk 3]
```diff
@@ -386,14 +386,14 @@
       type: string
       default: "0o077"
       example: ~
-    - name: dataset_event_manager_class
+    - name: dataset_manager_class
       description: Class to use as dataset event manager.
```
Comment (jedcunningham) [inline airflow/config_templates/config.yml]:
```suggestion
      description: Class to use as dataset manager.
```

File: airflow/config_templates/config.yml (hunk 1 above)
Comment (jedcunningham) [reply airflow/config_templates/config.yml]: I did just commit the "dataset event manager" -> "dataset manager" change though.

Comment (blag) [reply airflow/config_templates/config.yml]: :+1: Now if I can find the way to dismiss this suggestion...

**Review by uranusjr — APPROVED** [review]

**Pull request merged**